lib/probes/http.js

```diff
@@ -53,34 +75,40 @@ function patchClient (module, protocol) {
       span.async = true
 
       // Parse options object
-      if (typeof options === 'string') {
-        options = url.parse(options)
+      /*
+      if (typeof urlstring === 'string') {
+        urlopts = url.parse(urlstring)
       }
       const parsed = extend({}, options)
+      // */
 
       // Add X-Trace header to trace hops
       options.headers = options.headers || {}
       options.headers['X-Trace'] = span.events.entry.toString()
 
       // Set default protocol
-      parsed.protocol = parsed.protocol || protocol + ':'
+      options.protocol = options.protocol || protocol + ':'
 
+      let filtered = options
```

Comment:
as I understand `let` it creates a reference, so 'filtered' references 'options'
on line 102 'filtered' may become a copy of 'options'
and then both are used to extract entry event info.
I don't quite understand why a reference and sometimes a copy is needed, since 'options' is scoped to this block. Maybe because of refactoring it got like this.